package/amazon-linux-packaging.spec

```diff
@@ -0,0 +1,149 @@
+# dependency, it will look in the vendor/ folder instead of accessing the 
+# internet, enabling a completely offline build using the pre-vendored 
+# dependencies.
+mkdir -p .cargo
+cat > .cargo/config.toml <<'EOF'
+[source.crates-io]
+replace-with = "vendored-sources"
+
+[source.vendored-sources]
+directory = "vendor"
+EOF
+
+%build
+# FORTIFY_SOURCE throws warnings when optimization is missing, -Werror makes them fatal
+# This suppresses those warnings and ensures CMake can detect pthread properly
+export CFLAGS="${CFLAGS:-%{optflags}} -O2 -Wno-error=cpp"
+export CMAKE_C_FLAGS="$CFLAGS"
+
+# This compiles the Rust project in optimized release mode (--release) while 
+# preventing any network access (--offline) and forbidding changes to the 
+# dependency lock file (--frozen), ensuring a reproducible build using only the 
+# pre-vendored dependencies.
+cargo build --release --frozen --offline
+
+# Fedora packaging guidlines require a manifest that lists the names and versions of all
+# crates in the vendor tarball.  They also note that this file must be added as a %license file. Macros which do this automatically are not available on all platforms.
+grep -E '^(name|version) = ' Cargo.lock \
+  | sed -E 's/^(name = |version = )"//;s/"$//' \
+  | paste - - | tr '\t' '-' > cargo-vendor.txt
+
+%install
+
+install -d "%{buildroot}/opt/aws/mountpoint-s3/bin"
+install -d "%{buildroot}/usr/bin"
+install -d "%{buildroot}/usr/sbin"
+
+install -D -m0755 target/release/mount-s3 \
+    "%{buildroot}/opt/aws/mountpoint-s3/bin/mount-s3"
+
+ln -sf /opt/aws/mountpoint-s3/bin/mount-s3 "%{buildroot}/usr/bin/mount-s3"
+ln -sf /opt/aws/mountpoint-s3/bin/mount-s3 "%{buildroot}/usr/sbin/mount.mount-s3"
+
+install -D -m0644 %{SOURCE2} "%{buildroot}/opt/aws/mountpoint-s3/LICENSE"
+install -D -m0644 %{SOURCE3} "%{buildroot}/opt/aws/mountpoint-s3/NOTICE"
+install -D -m0644 %{SOURCE4} "%{buildroot}/opt/aws/mountpoint-s3/THIRD_PARTY_LICENSES"
+echo "%{version}" > "%{buildroot}/opt/aws/mountpoint-s3/VERSION"
+install -D -m0644 cargo-vendor.txt "%{buildroot}/opt/aws/mountpoint-s3/cargo-vendor.txt"
+
+%if %{with check}
+%check
+# We reuse the same optimized flags as in the  %build section
+export CFLAGS="%{optflags} -O2 -Wno-error=cpp"
+export CMAKE_C_FLAGS="$CFLAGS"
+
+# ulimit -n 4096 increases the maximum number of open file descriptors to 4096 
+# because Mountpoint-S3's tests open many files simultaneously and would fail with
+# the default lower limit.
+ulimit -n 4096
+cargo test --release --frozen --offline -- \
+    --skip mnt::test::mount_unmount \
+    --skip unmount_no_send
+%endif
+
+%files
+/opt/aws/mountpoint-s3
+/opt/aws/mountpoint-s3/LICENSE
+/opt/aws/mountpoint-s3/NOTICE
+/opt/aws/mountpoint-s3/THIRD_PARTY_LICENSES
+/opt/aws/mountpoint-s3/VERSION
+/opt/aws/mountpoint-s3/bin
+/opt/aws/mountpoint-s3/bin/mount-s3
+/opt/aws/mountpoint-s3/cargo-vendor.txt
+/usr/bin/mount-s3
+/usr/sbin/mount.mount-s3
+
+%changelog
```

Comment:
Is this changelog meant to be for this rpm or for Mountpoint itself?

Reply:
Can we just provide link to our changelog on GitHub here instead? Do we really need to duplicate changelog here?